pkg/service/did/key.go

```diff
@@ -29,29 +31,42 @@ func (h *keyDIDHandler) CreateDID(request CreateDIDRequest) (*CreateDIDResponse,
 	if err != nil {
 		return nil, errors.Wrap(err, "could not create did:key")
 	}
-	privKeyBase58, err := privateKeyToBase58(privKey)
-	if err != nil {
-		return nil, errors.Wrap(err, "could not encode private key as base58")
-	}
 
 	// expand it to the full doc for storage
 	expanded, err := doc.Expand()
 	if err != nil {
 		return nil, errors.Wrap(err, "error generating did:key document")
 	}
 
-	// store it
+	// store metadata in DID storage
+	id := doc.ToString()
 	storedDID := storage.StoredDID{
-		DID:              *expanded,
-		PrivateKeyBase58: privKeyBase58,
+		ID:  id,
+		DID: *expanded,
 	}
 	if err := h.storage.StoreDID(storedDID); err != nil {
 		return nil, errors.Wrap(err, "could not store did:key value")
 	}
 
+	// store private key in key storage
+	keyStoreRequest := keystore.StoreKeyRequest{
+		ID:         id,
+		Type:       request.KeyType,
+		Controller: id,
+		Key:        privKey,
+	}
+	if err := h.keyStore.StoreKey(keyStoreRequest); err != nil {
+		return nil, errors.Wrap(err, "could not store did:key private key")
+	}
+
+	privKeyBase58, err := privateKeyToBase58(privKey)
```

Comment:
Should this happen before you store the key?

Because there could be a case where you store a key but you can't convert it

Reply:
yeah, good call out